src/Avalonia.Controls/AppBuilderBase.cs

```diff
@@ -21,33 +22,48 @@ public abstract class AppBuilderBase<TAppBuilder> where TAppBuilder : AppBuilder
         /// <summary>
         /// Gets or sets a method to call the initialize the runtime platform services (e. g. AssetLoader)
         /// </summary>
-        public Action RuntimePlatformServices { get; set; }
+        public Action RuntimePlatformServicesInitializer { get; private set; }
 
         /// <summary>
         /// Gets or sets the <see cref="Application"/> instance being initialized.
         /// </summary>
-        public Application Instance { get; set; }
+        public Application Instance { get; protected set; }
 
         /// <summary>
         /// Gets or sets a method to call the initialize the windowing subsystem.
         /// </summary>
-        public Action WindowingSubsystem { get; set; }
+        public Action WindowingSubsystemInitializer { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the name of the windowing subsystem to use.
+        /// </summary>
+        public string WindowingSubsystemName { get; set; }
```

Comment:
Looking some more at the code, it looks like the name isn't for *selection*, it's a descriptor of the subsystem initializer that has been assigned. In that case, it should probably have a private setter, and be passed to `UseWindowingSubsystem` etc.

Or maybe an alernative would be a new class, say `SubsystemInitializer` which has initializer and name properties.